Add -m flag which allows to modify branch together with Jira issue summary

macanton opened the pull request:
## Summary
Current PR adds possibility to modify current branch issue together with current branch name

## Usage
```
got -m
Enter Jira issue summary: 
```

## Checklist
- checkout to branch which is already linked to Jira issue
- execute `got -m`
- enter new Jira issue summary
- check that issue updated and that branch name updated

- checkout to branch which is already linked to several Jira issues
- execute `got -m`
- enter new Jira issue summary
- check that first issue updated and that branch name updated

- checkout to branch wich is not linked to jira issue
- execute `got -m`
- enter new Jira issue summary
- should return an error

macanton committed:
Add -m flag which allows to modify branch together with Jira issue summary

Den1k22 commented on pkg/git/commands.go:
```diff
@@ -62,9 +62,36 @@ func CheckoutNewBranch(branchName string) ([]byte, error) {
+}
+
+// GetCurrentBranch returns current git branch
+func GetCurrentBranch() (string, error) {
```
What do you think to be more specific and call this function: `GetCurrentBranchName`

macanton replied:
Sounds good. With that change we can have a set of functions that operate with `Branch struct` in future

Den1k22 commented on main.go:
```diff
@@ -84,6 +86,43 @@ func createJiraTicketAndCheckBranch() {
+		printErrorToConsole(err)
+		return
+	}
+	printInfoToConsole(fmt.Sprintf("Jira issue summary updated to '%s'", config.Options.Summary))
```
What about to use `summary` instead of `config.Options.Summary`

macanton committed:
Refactor function name

- GetCurrentBranchName allows to define Branch struct
  in future

macanton committed:
Change source of Jira issue updated summary in printed notification

- jira.UpdateIssueSummary returns actually updated value which should
  be displayed to user. That allows to enhance UpdateIssueSummary in future
  with summary modification before sending request
